File: 19주차/이연상/귤 고르기.py
# [PRG] 귤 고르기

def solution(k, tangerine):
    answer = 0
    dic = {}
    for i in tangerine:
        if i in dic:
            dic[i] +=1
        else:
            dic[i] = 1
    
    temp = sorted(dic.items(), key = lambda x : x[1], reverse=True)
    
    for i in temp:
        if k <= 0:
            return answer
        k -= i[1]
        answer += 1
        
    
    return answer

File: 43주차/이연상/귤 고르기.java
// [PRG] 귤 고르기

import java.util.*;

class Solution {
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        Arrays.sort(tangerine);
        int[] arr = new int[tangerine[tangerine.length - 1] + 1];
        for(int i = 0; i < tangerine.length; i++) {
            arr[tangerine[i]] += 1;
        }
        Arrays.sort(arr);
        for(int i = arr.length - 1; i > 0; i--) {
            if(k == 0)  break;
            answer += 1;
            
            for(int j = arr[i]; j > 0; j--) {
                k -= 1;
                if(k == 0) break;
            }
        }
        
        return answer;
    }
}